src/Controls/src/Core/Platform/GestureManager/GesturePlatformManager.Windows.cs

```diff
@@ -747,11 +751,13 @@ void UpdatingGestureRecognizers()
 			_container.PointerPressed += OnPgrPointerPressed;
 			_container.PointerReleased += OnPgrPointerReleased;
 
-			bool hasSwipeGesture = gestures.GetGesturesFor<SwipeGestureRecognizer>().GetEnumerator().MoveNext();
-			bool hasPinchGesture = gestures.GetGesturesFor<PinchGestureRecognizer>().GetEnumerator().MoveNext();
-			bool hasPanGesture = gestures.GetGesturesFor<PanGestureRecognizer>().GetEnumerator().MoveNext();
+			bool hasSwipeGesture = gestures.HasAnyGesturesFor<SwipeGestureRecognizer>();
+			bool hasPinchGesture = gestures.HasAnyGesturesFor<PinchGestureRecognizer>();
+			bool hasPanGesture = gestures.HasAnyGesturesFor<PanGestureRecognizer>();
```

Comment:
So one thing to consider is this would still iterate over the gestures list three times. How many are normally in this list? If it's usually like 1-5, that might be OK.